stylix/mk-target.nix

```
@@ -0,0 +1,96 @@
+# See docs/src/modules.md#mkTarget for full explanation
+{
+  # Used when generating options; `stylix.targets.${name}`
+  name,
+
+  # Used when generating enable option; `stylix.targets.${name}.enable` has a
+  # description of "Whether to enable theming for ${humanName}."
+  humanName,
+
+  # Whether the target should be auto-enabled
+  autoEnable ? true,
+
+  # Attributes will be mapped to `stylix.targets.${name}.${option}`
+  extraOptions ? { },
+
+  # Function(s) should take the Stylix attributes that it needs (e.g. `colors`,
+  # `image`, `fonts`) and `cfg` if necessary. This `cfg` is an alias for
+  # `config.stylix.targets.${name}`. Function(s) should return an attribute set
+  # to be merged into `config`. If any of the Stylix attributes needed by a/the
+  # function are `null`, AKA unset, the configuration will not be merged.
+  configElements ? [ ],
+
+  # Same format as functions under `configElements`. Merged regardless of whether
+  # requested Stylix options are `null`. Avoid use when possible.
+  generalConfig ? null,
+}:
+# Note: we only return a plain module so that calling code can use module args, while using this as their module body
+let
+  module =
+    { config, lib, ... }:
+    let
+      cfg = config.stylix.targets.${name};
+
+      # Get the function's de-structured arg names as a list
+      functionArgNames =
+        fn:
+        lib.pipe fn [
+          lib.functionArgs
+          builtins.attrNames
+        ];
+
+      # Get the Stylix attrs (option values) requested by the function's de-structured args
+      getStylixAttrs =
+        fn:
+        lib.genAttrs (functionArgNames fn) (
+          arg:
+          if arg == "cfg" then
+            cfg
+          else if arg == "colors" then
+            config.lib.stylix.colors
+          else
+            config.stylix.${arg}
+              or (throw "stylix: Stylix arg was requested by mk-target but does not exist: ${arg}")
+        );
+
+      # Call the config element function with the requested stylix attr values
+      # Returns a config definition
```

I think the implementation "detail" of specifically requesting Stylix arguments is too specific for the function documentation:

```suggestion
      # Call the configuration function with its required arguments.
```

---

> too specific for the function documentation

Note that `#` and `/* */` comments are not documentation, they are normal comments intended to explain why something is done, clarify impl details, etc.

i.e. notes to people hacking on `mkTarget`, not notes for people using `mkTarget`.

Documentation should go in a doc-comment `/** */` and/or markdown files.

---

Yes, but even for internal function documentations geared towards people hacking on `mkTarget`, it can be confusing when one function assumes too much external behavior.

However, I am also fine keeping the "Stylix arguments" detail:

```suggestion
      # Call the configuration function with its required Stylix arguments.
```